SharedPackages/DataBrokerProtectionCore/Sources/DataBrokerProtectionCore/Operations/EmailConfirmationJob.swift

```diff
@@ -0,0 +1,327 @@
+    private(set) weak var errorDelegate: EmailConfirmationErrorDelegate? // Internal read-only to enable mocking
+    private let jobDependencies: EmailConfirmationJobDependencyProviding
+
+    private let id = UUID()
+    private var _isExecuting = false
+    private var _isFinished = false
+
+    private static let maxRetries = 3
+
+    deinit {
+        Logger.dataBrokerProtection.log("✉️ Deinit EmailConfirmationJob: \(String(describing: self.id.uuidString))")
+    }
+
+    public init(jobData: OptOutEmailConfirmationJobData,
+                showWebView: Bool,
+                errorDelegate: EmailConfirmationErrorDelegate?,
+                jobDependencies: EmailConfirmationJobDependencyProviding) {
+        self.jobData = jobData
+        self.showWebView = showWebView
+        self.errorDelegate = errorDelegate
+        self.jobDependencies = jobDependencies
+        super.init()
+    }
+
+    public override func start() {
+        if isCancelled {
+            finish()
+            return
+        }
+
+        willChangeValue(forKey: #keyPath(isExecuting))
+        _isExecuting = true
+        didChangeValue(forKey: #keyPath(isExecuting))
+
+        main()
+    }
+
+    public override var isAsynchronous: Bool {
+        return true
+    }
+
+    public override var isExecuting: Bool {
+        return _isExecuting
+    }
+
+    public override var isFinished: Bool {
+        return _isFinished
+    }
+
+    public override func main() {
+        Task {
+            await runJob()
+            finish()
+        }
+    }
+
+    private func runJob() async {
+        guard let emailConfirmationLink = jobData.emailConfirmationLink,
+              let linkURL = URL(string: emailConfirmationLink) else {
+            await handleError(EmailError.invalidEmailLink)
+            return
+        }
+
+        // Fetch the broker data
+        guard let broker = try? jobDependencies.database.fetchBroker(with: jobData.brokerId) else {
+            await handleError(DataBrokerProtectionError.dataNotInDatabase)
+            return
+        }
+
+        // Fetch the extracted profile
+        guard let extractedProfileData = try? jobDependencies.database.fetchExtractedProfile(with: jobData.extractedProfileId) else {
+            await handleError(DataBrokerProtectionError.dataNotInDatabase)
+            return
+        }
+
+        let extractedProfile = extractedProfileData.profile
+
+        var attemptCount = jobData.emailConfirmationAttemptCount
+
+        let stageDurationCalculator = DataBrokerProtectionStageDurationCalculator(
+            dataBroker: broker.url,
+            dataBrokerVersion: broker.version,
+            handler: jobDependencies.pixelHandler,
+            vpnConnectionState: jobDependencies.vpnBypassService?.connectionStatus ?? "unknown",
+            vpnBypassStatus: jobDependencies.vpnBypassService?.bypassStatus.rawValue ?? "unknown"
+        )
+
+        while attemptCount < Self.maxRetries {
```

Comment:
I was imagining retries would work differently. If we retry them straight away, if they failed the first time it seems likely the outcome will be the same this time.

I think it would be better if a job consisted of a single attempt, and aren't retried until the queue is populated again